server/service/http/transaction_service.rs

```diff
@@ -1070,22 +1075,26 @@ impl TransactionService {
         } else {
             let named_outputs = pipeline.rows_positions().unwrap();
             let descriptor: StreamQueryOutputDescriptor = named_outputs.clone().into_iter().sorted().collect();
-
-            let encode_pipeline_structure_result = pipeline
-                .pipeline_structure()
-                .map(|qs| encode_analyzed_pipeline_for_studio(&*snapshot, &type_manager, qs))
-                .transpose();
-            let include_involved_blocks = IncludeInvolvedBlocks::build(pipeline.pipeline_structure());
-            let pipeline_structure_response = match encode_pipeline_structure_result {
-                Ok(structure_opt) => structure_opt,
-                Err(typedb_source) => {
-                    respond_error_and_return_break!(
-                        responder,
-                        TransactionServiceError::PipelineExecution {
-                            typedb_source: PipelineExecutionError::ConceptRead { typedb_source }
-                        }
-                    );
-                }
+            let (encoded_structure, include_involved_blocks) = if query_options.include_query_structure {
+                let encode_pipeline_structure_result = pipeline
+                    .pipeline_structure()
+                    .map(|qs| encode_analyzed_pipeline_for_studio(&*snapshot, &type_manager, qs))
+                    .transpose();
+                let pipeline_structure_response = match encode_pipeline_structure_result {
+                    Ok(structure_opt) => structure_opt,
+                    Err(typedb_source) => {
+                        respond_error_and_return_break!(
+                            responder,
+                            TransactionServiceError::PipelineExecution {
+                                typedb_source: PipelineExecutionError::ConceptRead { typedb_source }
+                            }
+                        );
+                    }
+                };
+                let include_involved_blocks = IncludeInvolvedBlocks::build(pipeline.pipeline_structure());
+                (pipeline_structure_response, include_involved_blocks)
```

Comment:
Can the body here be a function? Or even the entire `if`. It seems to repeat three times across two modules.